### ISSUE TITLE
[Package Issue]: Microsoft.RemoteDesktopClient

### ISSUE THREAD

**Issue by johnterickson** [microsoft/winget-pkgs#102249]
### Please confirm these before moving forward

- [X] I have searched for my issue and not found a work-in-progress/duplicate/resolved issue.
- [X] I have not been informed if the issue is resolved in a preview version of the winget client.

### Category of the issue

Installation issue.

### Brief description of your issue

`Installer hash does not match` 

The manifest for `1.2.4066.0` points to `https://go.microsoft.com/fwlink/?linkid=2068602` but that downloads as `RemoteDesktop_1.2.4157.0_x64.msi`.

@FaserF @mdanish-kh from https://github.com/microsoft/winget-pkgs/pull/101107

### Steps to reproduce

winget upgrade Microsoft.RemoteDesktopClient --verbose

### Actual behavior

```
Found Remote Desktop [Microsoft.RemoteDesktopClient] Version 1.2.4066.0
This application is licensed to you by its owner.
Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
Downloading https://go.microsoft.com/fwlink/?linkid=2068602
  33.6 MB
Installer hash does not match; this cannot be overridden when running as admin
```

### Expected behavior

Package with correct hash is downloaded.

### Environment

```raw
C:\Users\jerick>winget --info
Windows Package Manager v1.4.10173
Copyright (c) Microsoft Corporation. All rights reserved.

Windows: Windows.Desktop v10.0.22621.1555
System Architecture: X64
Package: Microsoft.DesktopAppInstaller v1.19.10173.0

Logs: %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\DiagOutputDir

User Settings: %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\settings.json
```

**Comment by johnterickson** [comment]
Possibly related to https://github.com/microsoft/winget-pkgs/issues/99081

**Comment by BrandonWanHuanSheng** [comment]
I can't find x86 because x64 point to this file https://query.prod.cms.rt.microsoft.com/cms/api/am/binary/RW11eHT How about x86 URL?

**Comment by ItzLevvie** [comment]
> I can't find x86 because x64 point to this file https://query.prod.cms.rt.microsoft.com/cms/api/am/binary/RW11eHT How about x86 URL?

You can find these installers at https://learn.microsoft.com/en-us/azure/virtual-desktop/whats-new-client-windows#supported-client-versions

**Comment by BrandonWanHuanSheng** [comment]
It look like the old URL link is still work but I can't find this https://go.microsoft.com/fwlink/?linkid=2068602 URL on Azure Page.

**Comment by BrandonWanHuanSheng** [comment]
That go.microsoft.com sometimes points to its Azure Latest Version.

**Comment by BrandonWanHuanSheng** [comment]
Preparing to correct the URL before the upgrade. Later on, I will pull up another request. Thank you.

**Comment by Trenly** [comment]
Resolved by #102433
